src/extensions/qdatabase.c

```diff
@@ -86,13 +95,204 @@
 #define Q_MYSQL_OPT_CONNECT_TIMEOUT    (10)
 #define Q_MYSQL_OPT_READ_TIMEOUT       (30)
 #define Q_MYSQL_OPT_WRITE_TIMEOUT      (30)
-#endif
+#endif /* ENABLE_MYSQL */
+
+#ifdef ENABLE_PGSQL
+#include "libpq-fe.h"
+
+/*
+Wrap the low-level functions of the libpq library.
+Provide interfaces with naming convention similar to mysql_xyz().
+*/
+
+typedef enum {
+    PQ_UNKNOWN,
+    PQ_PING,
+    PQ_QUERY,
+    PQ_UPDATE,
+    PQ_TRANSACTION,
+} pgquery_t;
+
+typedef struct {
+    char   *emsg;
+    PGconn *pgconn;
+
+    PGresult *pgresult;
+    int rows;
+    int cols;
+    int cursor;
+} pgsql_t;
```

Comment:
We have qdb_s struct already for this. Having a separate struct doesn't seem to be necessary.